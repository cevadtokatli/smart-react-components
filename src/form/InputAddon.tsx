import styled from "styled-components"
import ElementProps from "../element/Props"
import Div from "../element/Div"

interface Props extends ElementProps {
    type: string
    shape: string
    absolute: boolean
}

export default styled(Div)<Props>(({theme,type,shape,absolute}) => `
    overflow: hidden;
    flex: 0 0 auto;
    display: inline-flex;

    ${!absolute ? `
        ${type == "left"
        ? `
            border-top-left-radius: ${theme.src.form.inputRadius[shape]};
            border-bottom-left-radius: ${theme.src.form.inputRadius[shape]};
        ` : `
            border-top-right-radius: ${theme.src.form.inputRadius[shape]};
            border-bottom-right-radius: ${theme.src.form.inputRadius[shape]};
        `}
    ` : `
        position: absolute;
    `}
`)