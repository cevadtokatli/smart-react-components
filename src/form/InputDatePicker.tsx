import React from "react"
import InputProps from "./InputProps"
import {Value, Language} from "../types/date-picker"
import {JSXElementProps} from "../props"
import DV from "../default-value"
import * as PT from "../constants/date-picker/picker-type"
import * as DT from "../constants/date-picker/date-type"
import * as HT from "../constants/date-picker/hour-type"
import DEFAULT_LANGUAGE from "../constants/date-picker/default-language"
import DatePickerHelper from "../helper/DatePickerHelper"
import StringHelper from "../helper/StringHelper"
import DOMHelper from "../helper/DOMHelper"
import useInputOutput from "./UseInputOutput"
import FixedBox from "../fixed-box/FixedBox"
import InputElement from "./InputElement"
import DatePicker from "../date-picker/DatePicker"

export interface Props extends InputProps<Value> {
    hint?: never
    error?: never
    pickerSize?: string
    pickerSizeSm?: string
    pickerSizeMd?: string
    pickerSizeLg?: string
    pickerSizeXl?: string
    pickerElementProps?: JSXElementProps
    pickerThemeType?: string
    pickerType?: number
    dateType?: number
    hourType?: number
    showButtons?: boolean
    weekStartDay?: number
    orientation?: boolean
    lang?: Language
}

const InputDatePicker: React.FC<Props> = ({size="default",sizeSm,sizeMd,sizeLg,sizeXl,containerProps=DV.JSX_ELEMENT_PROPS,inputProps=DV.JSX_ELEMENT_PROPS,labelProps=DV.JSX_ELEMENT_PROPS,label,value,setValue,type,fill,border,shape="default",leftAddon,leftAddonProps=DV.JSX_ELEMENT_PROPS,leftAddonAbsolute,rightAddon,rightAddonProps=DV.JSX_ELEMENT_PROPS,rightAddonAbsolute,pickerSize,pickerSizeSm,pickerSizeMd="small",pickerSizeLg="default",pickerSizeXl,pickerElementProps,pickerThemeType,pickerType=PT.DATE_HOUR,dateType=DT.MDY,hourType=HT.$12,showButtons=true,weekStartDay,orientation,lang=DEFAULT_LANGUAGE}) => {
    const getValue$ = (): string => {
        const valueObject = DatePickerHelper.getValueObject(value, pickerType, hourType)
        let result = ""

        if(valueObject) {
            if(valueObject.date) {
                if(dateType == DT.DMY)
                    result += `${StringHelper.renderNumber(valueObject.date.day, 2, "0")}/${StringHelper.renderNumber(valueObject.date.month+1, 2, "0")}/${StringHelper.renderNumber(valueObject.date.year, 4, "0")}`
                else
                    result += `${StringHelper.renderNumber(valueObject.date.month+1, 2, "0")}/${StringHelper.renderNumber(valueObject.date.day, 2, "0")}/${StringHelper.renderNumber(valueObject.date.year, 4, "0")}`
            }

            if(valueObject.hour) {
                if(result)
                    result += " "

                result += `${StringHelper.renderNumber(valueObject.hour.hour, 2, "0")}:${StringHelper.renderNumber(valueObject.hour.minute, 2, "0")}`

                if(hourType == HT.$12)
                    result += ` ${lang[valueObject.hour.am ? "am" : "pm"]}`
            }
        }

        return result
    }
    
    const [status, setStatus] = React.useState<boolean>(false)
    const [value$, setValue$] = React.useState<string>(() => getValue$())
    const inputEl = React.useRef<HTMLInputElement>(null)
    const valueInit = React.useRef<boolean>(false)

    /**
     * Converts the given string value to the original value format.
     * 
     * @param _value
     */
    const convertValue = (_value:string): Value => {
        if(dateType == DT.DMY) {
            const valueSplit = _value.split("/")
            if(valueSplit.length == 3)
                _value = `${valueSplit[1]}/${valueSplit[0]}/${valueSplit[2]}`
        }

        if(typeof value === "string")
            return _value
        else if(value.constructor == Date)
            return new Date(_value)
        else
            return DatePickerHelper.getValueObject(_value, pickerType, hourType)
    }

    /**
     * Prevents event, if it is mobile.
     * 
     * @param e 
     */
    const inputFocus = (e:React.FormEvent<HTMLInputElement>) => {
        if(DOMHelper.isMobile)
            (e.target as HTMLInputElement).blur()
    }

    /**
     * Sets value, if it is not mobile.
     * 
     * @param e 
     */
    const inputBlur = (e:React.FormEvent<HTMLInputElement>) => {
        if(!DOMHelper.isMobile)
            setValue(convertValue((e.target as HTMLInputElement).value))
    }


    /**
     * When enter is pressed, changes value.
     * 
     * @param e
     */
    const inputKeyDown = (e:React.KeyboardEvent<HTMLInputElement>) => {
        if(e.keyCode == 13)
            setValue(convertValue((e.target as HTMLInputElement).value))
    }

    /**
     * Closes box.
     */
    const close = () => setStatus(false)

    React.useEffect(() => {
        if(!valueInit.current)
            valueInit.current = true
        else
            setValue$(getValue$())
    }, [value, pickerType, hourType])

    React.useEffect(() => {
        inputEl.current.value = value$
    }, [value$])

    return (
        <FixedBox boxProps={{width:["auto", true]}} status={status} setStatus={setStatus}>
            {useInputOutput({
                element: <InputElement {...inputProps} ref={inputEl} defaultValue={value$} tabIndex={-1} onFocus={inputFocus} onBlur={inputBlur} onKeyDown={inputKeyDown} size$={size} size$Sm={sizeSm} size$Md={sizeMd} size$Lg={sizeLg} size$Xl={sizeXl} type$={type} fill$={fill} border$={border} shape={shape} leftAddon={leftAddon ? true : false} leftAddonAbsolute={leftAddonAbsolute} rightAddon={rightAddon ? true : false} rightAddonAbsolute={rightAddonAbsolute} />,
                containerProps,inputProps,labelProps,label,shape,leftAddon,leftAddonProps,leftAddonAbsolute,rightAddon,rightAddonProps,rightAddonAbsolute
            })}
            <DatePicker size={pickerSize} sizeSm={pickerSizeSm} sizeMd={pickerSizeMd} sizeLg={pickerSizeLg} sizeXl={pickerSizeXl} elementProps={pickerElementProps} value={value} setValue={setValue} type={pickerThemeType} pickerType={pickerType} hourType={hourType} showCloseIcon={true} showButtons={showButtons} weekStartDay={weekStartDay} orientation={orientation} lang={lang} onClose={close} onSave={close} onCancel={close} />
        </FixedBox>
    ) 
}

export default InputDatePicker