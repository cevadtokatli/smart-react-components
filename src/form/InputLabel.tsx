import styled from "styled-components"
import ElementProps from "../element/Props"
import Label from "../element/Label"

interface Props extends ElementProps {
    disabled: boolean
}

export default styled(Label).attrs<Props>(({mb="2"}) => ({mb}))<Props>(({theme,disabled}) => `
    display: inline-block;
    word-wrap: break-word;
    word-break: break-word;

    ${disabled ? `
        opacity: ${theme.src.form.disabledOpacity};
        user-select: none;
    ` : ""}
`)