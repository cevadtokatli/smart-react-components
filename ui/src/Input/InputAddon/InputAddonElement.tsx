import Div from '@smart-react-components/core/Element/Div'
import Label from '@smart-react-components/core/Element/Label'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { PaletteProp, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'
import { OrderPosition } from '../../types'
import { toCSSValue } from '../../util/css'
import { getReverseOrderPosition } from '../../util/props'

interface Props extends
  StyledProps,
  ResponsiveProp<'inputSize', SizeProp> {
  addonPosition: OrderPosition
  hasBorder: boolean
  isDisabled: boolean
  isFocused: boolean
  isOutline: boolean
  isSeparated: boolean
  isSoft: boolean
  palette: PaletteProp
  shape?: ShapeProp
}

export default styled(Div).attrs<Props>(({ isSeparated }: Props) => ({
  ...(isSeparated && { as: Label }),
  getInputSize: (v, t) => `
    font-size: ${t.$.size.form.blockLabel[v].fontSize};
    padding: ${isSeparated ? t.$.size.input[v].padding.y : 0} ${toCSSValue(t.$.size.input[v].padding.x)(v => v / 2)};
  `,
}))<Props>(({ theme, addonPosition, hasBorder, isDisabled, isFocused, isOutline, isSeparated, isSoft, palette, shape }: Props) => `
  align-items: center;
  border-bottom-${OrderPosition[addonPosition].toLowerCase()}-radius: ${theme.$.radius.input[shape]};
  border-top-${OrderPosition[addonPosition].toLowerCase()}-radius: ${theme.$.radius.input[shape]};
  box-sizing: border-box;
  display: inline-flex;
  flex: 0 0 auto;
  justify-content: center;
  transition: 200ms 0s ease-in-out;
  transition-property: background, border, color, fill;

  ${isSeparated
    ? `
      align-self: flex-start;
    `
    : ''
  }

  ${hasBorder
    ? `
      border: solid 1px;
      border-${OrderPosition[getReverseOrderPosition(addonPosition)].toLowerCase()}: 0;
    `
    : ''
  }

  ${!isOutline
    ? `
      ${!isSoft
        ? `
          color: ${theme.$.palette[palette].font};
          fill: ${theme.$.palette[palette].font};

          ${(isSeparated || !isFocused)
            ? `
              background: ${theme.$.palette[palette].main};
              border-color: ${theme.$.palette[palette].dark};
            `
            : `
              background: ${theme.$.palette[palette].dynamicer};
              border-color: ${theme.$.palette[palette].dynamicest};
            `
          }
        `
        : `
          color: ${theme.$.palette[palette].softFont};
          fill: ${theme.$.palette[palette].softFont};

          ${(isSeparated || !isFocused)
            ? `
              background: ${theme.$.palette[palette].soft};
              border-color: ${theme.$.palette[palette].softDark};
            `
            : `
              background: ${theme.$.palette[palette].softDarker};
              border-color: ${theme.$.palette[palette].softDarkest};
            `
          }
        `
      }
    `
    : `
      background: ${theme.$.color.dynamic.background};
      border-color ${(isSeparated || !isFocused) ? theme.$.color.dynamic.accent : theme.$.palette[palette].main};
      color: ${theme.$.color.dynamic.font};
      fill: ${theme.$.color.dynamic.font};
    `
  }

  ${(!isSeparated && isDisabled)
    ? `
      opacity: ${theme.$.opacity.formDisabled};

      ${isOutline
        ? `
          background: ${theme.$.color.dynamic.accent};  
          color: ${theme.$.color.dynamic.gray};
        `
        : ''
      }
    `
    : ''
  }
`)
