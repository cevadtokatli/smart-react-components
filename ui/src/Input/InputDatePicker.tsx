import Div from '@smart-react-components/core/Element/Div'
import extractElementProps from '@smart-react-components/core/element-props'
import changeEvents, { ChangeEvents } from '@smart-react-components/core/element-props/change-events'
import focusEvents, { FocusEvents } from '@smart-react-components/core/element-props/focus-events'
import intrinsicStyledCoreProps, { IntrinsicStyledCoreProps } from '@smart-react-components/core/element-props/intrinsic-styled-core-props'
import keyboardEvents, { KeyboardEvents } from '@smart-react-components/core/element-props/keyboard-events'
import { ContentElement, JSXElementProps, PaletteProp, SetState, ShapeProp, SizeProps } from '@smart-react-components/core/types'
import React from 'react'
import FormBlockLabel from '../components/Form/FormBlockLabel'
import InputElement from '../components/Input/InputElement'
import InputWrapper from '../components/Input/InputWrapper'
import { DatePickerTimeType, DatePickerType } from '../constants/date-picker'
import useAddons from '../hooks/useAddons'
import { applyResponsiveStyledProp } from '../util/props'
import FixedBox from '../FixedBox'
import InputAddon from './InputAddon'
import DatePicker from '../DatePicker'

export { DatePickerType, DatePickerTimeType } from '../constants/date-picker'

export interface Props extends
  IntrinsicStyledCoreProps,
  SizeProps,
  ChangeEvents,
  FocusEvents,
  KeyboardEvents {
  elementProps?: JSXElementProps
  hasBorder?: boolean
  isBlock?: boolean
  isDisabled?: boolean
  isOutline?: boolean
  isReadOnly?: boolean
  isRequired?: boolean
  isSoft?: boolean
  label?: string
  leftAddon?: ContentElement
  onCancel?: (e: Event) => void
  onClose?: (e: Event) => void
  onSave?: (e: Event) => void
  palette?: PaletteProp
  pickerType?: DatePickerType
  placeholder?: string
  rightAddon?: ContentElement
  shape?: ShapeProp
  timeType?: DatePickerTimeType
  setValue: SetState<Date>
  value: Date
}

const InputDatePicker = React.forwardRef<HTMLInputElement, Props>((props, forwardRef) => {
  const ref = React.useRef<HTMLInputElement>(null)

  const [status, setStatus] = React.useState(() => false)

  const { leftAddon, rightAddon } = useAddons({
    Component: InputAddon,
    leftAddon: props.leftAddon,
    rightAddon: props.rightAddon,
    props: {
      cursorKey: 'selectBox',
      hasBorder: props.hasBorder,
      isDisabled: props.isDisabled,
      isFocused: status,
      isOutline: props.isOutline,
      isSoft: props.isSoft,
      palette: props.palette,
      shape: props.shape,
      size: props.size,
      sizeSm: props.sizeSm,
      sizeMd: props.sizeMd,
      sizeLg: props.sizeLg,
      sizeXl: props.sizeXl,
    },
  })

  const handleBlur = () => {
    const value = ref.current?.value

    if (value) {
      props.setValue(new Date(value))
    } else {
      props.setValue(null)
    }
  }

  React.useEffect(() => {
    ref.current.value = props.value?.toDateString() ?? ''
  }, [props.value])

  return (
    <FormBlockLabel
      {...extractElementProps(props, [intrinsicStyledCoreProps])}
      cursorKey="selectBox"
      formSize={props.size}
      formSizeSm={props.sizeSm}
      formSizeMd={props.sizeMd}
      formSizeLg={props.sizeLg}
      formSizeXl={props.sizeXl}
      isBlock={props.isBlock}
      isDisabled={props.isDisabled}
      palette={props.palette}
    >
      { props.label && <span>{props.label}</span> }
      <Div
        display="flex"
        flex="1 1 auto"
      >
        { leftAddon?.props?.isSeparated && leftAddon }
        <InputWrapper
          hasBorder={props.hasBorder}
          hasSeparatedLeftAddon={leftAddon?.props?.isSeparated}
          hasSeparatedRightAddon={rightAddon?.props?.isSeparated}
          isDisabled={props.isDisabled}
          isFocused={false}
          isOutline={props.isOutline}
          isSoft={props.isSoft}
          palette={props.palette}
          shape={props.shape}
        >
          { (leftAddon && !leftAddon?.props?.isSeparated) && leftAddon }
          <FixedBox
            elementProps={{
              ...applyResponsiveStyledProp(props, 'size', 'width', v => `$size.colorPicker.${v}.width`),
              background: '$color.dynamic.background',
              borderRadius: '$radius.colorPicker',
              boxShadow: '0 0 10px 1px {color.dynamic.shadow}',
              height: ['auto', true],
            }}
            maxWidth={-1}
            minWidth={-1}
            setStatus={setStatus}
            status={status}
          >
            <InputElement
              {...(!props.isDisabled && extractElementProps(props, [changeEvents, focusEvents, keyboardEvents]))}
              {...(!props.isDisabled && {
                onBlur: handleBlur,
              })}
              {...(props.isDisabled && { disabled: true })}
              {...(props.isReadOnly && { readOnly: true })}
              {...(props.isRequired && { required: true })}
              {...(typeof props.placeholder !== 'undefined' && { placeholder: props.placeholder })}
              defaultValue={props.value?.toDateString() ?? ''}
              hasLeftAddon={!!leftAddon}
              hasRightAddon={!!rightAddon}
              inputSize={props.size}
              inputSizeSm={props.sizeSm}
              inputSizeMd={props.sizeMd}
              inputSizeLg={props.sizeLg}
              inputSizeXl={props.sizeXl}
              isDisabled={props.isDisabled}
              isFocused={status}
              isOutline={props.isOutline}
              isSoft={props.isSoft}
              palette={props.palette}
              ref={forwardRef ?? ref}
            />
            <DatePicker
              {...applyResponsiveStyledProp(props, 'size', 'size', v => v)}
              hasCloseIcon
              onClose={() => setStatus(false)}
              isSoft={props.isSoft}
              palette={props.palette}
              pickerType={props.pickerType}
              timeType={props.timeType}
              setValue={props.setValue}
              value={props.value}
            />
          </FixedBox>
          { (rightAddon && !rightAddon?.props?.isSeparated) && rightAddon }
        </InputWrapper>
        { rightAddon?.props?.isSeparated && rightAddon }
      </Div>
    </FormBlockLabel>
  )
})

InputDatePicker.defaultProps = {
  hasBorder: true,
  isOutline: true,
  palette: 'primary',
  shape: 'rectangle',
  size: 'medium',
}

export default InputDatePicker
