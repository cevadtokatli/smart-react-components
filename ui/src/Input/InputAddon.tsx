import extractElementProps from '@smart-react-components/core/element-props'
import clickEvents, { ClickEvents } from '@smart-react-components/core/element-props/click-events'
import { ContentElement, JSXElementProps, PaletteProp, Partial, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import React from 'react'
import { OrderPosition } from '../types'
import InputAddonElement from '../components/Input/InputAddonElement'

export interface Props extends
  Partial<ResponsiveProp<'size', SizeProp>>,
  ClickEvents {
  children: ContentElement
  elementProps?: JSXElementProps
  isOutline?: boolean
  isSeparated?: boolean
  isSoft?: boolean
  palette?: PaletteProp
}

interface PrivateProps {
  hasBorder: boolean
  isDisabled: boolean
  isFocused: boolean
  position: OrderPosition
  shape: ShapeProp
}

const InputAddon = React.forwardRef<HTMLDivElement, Props>((props: Props & PrivateProps, forwardRef) => (
  <InputAddonElement
    addonPosition={props.position}
    hasBorder={props.hasBorder}
    inputSize={props.size}
    inputSizeSm={props.sizeSm}
    inputSizeMd={props.sizeMd}
    inputSizeLg={props.sizeLg}
    inputSizeXl={props.sizeXl}
    isDisabled={props.isDisabled}
    isFocused={props.isFocused}
    isOutline={props.isOutline}
    isSeparated={props.isSeparated}
    isSoft={props.isSoft}
    palette={props.palette}
    ref={forwardRef}
    shape={props.shape}
    {...extractElementProps(props, [clickEvents])}
    {...props.elementProps}
  >
    {props.children}
  </InputAddonElement>
))

InputAddon.defaultProps = {
  elementProps: {},
}

InputAddon.displayName = 'SRCAddon'

export default InputAddon
