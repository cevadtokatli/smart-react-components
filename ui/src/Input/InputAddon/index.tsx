import extractElementProps from '@smart-react-components/core/element-props'
import clickEvents, { ClickEvents } from '@smart-react-components/core/element-props/click-events'
import { ContentElement, PaletteProp, Partial, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import React from 'react'
import { OrderPosition } from '../../types'
import InputAddonElement from './InputAddonElement'

export interface Props extends
  Partial<ResponsiveProp<'size', SizeProp>>,
  ClickEvents {
  children: ContentElement
  isOutline?: boolean
  isSeparated?: boolean
  isSoft?: boolean
  palette?: PaletteProp
}

interface PrivateProps {
  hasBorder: boolean
  isDisabled: boolean
  isFocused: boolean
  position: OrderPosition
  shape: ShapeProp
}

const InputAddon: React.FC<Props> = (props: Props & PrivateProps) => (
  <InputAddonElement
    addonPosition={props.position}
    hasBorder={props.hasBorder}
    inputSize={props.size}
    inputSizeSm={props.sizeSm}
    inputSizeMd={props.sizeMd}
    inputSizeLg={props.sizeLg}
    inputSizeXl={props.sizeXl}
    isDisabled={props.isDisabled}
    isFocused={props.isFocused}
    isOutline={props.isOutline}
    isSeparated={props.isSeparated}
    isSoft={props.isSoft}
    palette={props.palette}
    shape={props.shape}
    {...extractElementProps(props, [clickEvents])}
  >
    {props.children}
  </InputAddonElement>
)

InputAddon.displayName = 'SRCAddon'

export default InputAddon
