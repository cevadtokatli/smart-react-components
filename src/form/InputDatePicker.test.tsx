import React from "react"
import renderTheme from "../test/renderTheme"
import mountTheme from "../test/mountTheme"
import InputDatePicker from "./InputColorPicker"
import InputLabel from "./InputLabel"
import InputAddon from "./InputAddon"

describe("<InputDatePicker />", () => {
    it("should match snapshot", () => {
        const wrapper = renderTheme(<InputDatePicker label="Label" />)
        expect(wrapper).toMatchSnapshot()
    })
    
    it("should call setValue", () => {
        const setValue = jest.fn()
        const wrapper = mountTheme(<InputDatePicker setValue={setValue} />)
        wrapper.find("input").simulate("blur")
        expect(setValue).toHaveBeenCalled()
    })

    it("should render label", () => {
        const wrapper = mountTheme(<InputDatePicker label="Label" />)
        expect(wrapper.find(InputLabel).exists()).toBe(true)
    })

    it("should render left addon", () => {
        const wrapper = mountTheme(<InputDatePicker leftAddon="Left Addon" />)
        expect(wrapper.find(InputAddon).exists()).toBe(true)
    })

    it("should render right addon", () => {
        const wrapper = mountTheme(<InputDatePicker rightAddon="Right Addon" />)
        expect(wrapper.find(InputAddon).exists()).toBe(true)
    })
})