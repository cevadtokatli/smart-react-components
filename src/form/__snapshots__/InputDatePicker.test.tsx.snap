// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputDatePicker /> should match snapshot 1`] = `
<div
  class="sc-AxjAm sc-Axmtr jtvZOT"
>
  <label
    class="sc-AxmLO sc-fzozJi gncWLU"
  >
    Label
  </label>
  <input
    class="sc-fzoLsD sc-fzpans cKluuF"
    display="block"
    font-family="inherit"
    shape="default"
    tabindex="-1"
    width="100%"
  />
</div>
`;
